File: processor/src/test/resources/scenarios/cassandracodecs/NonAccessibleCassandraCodec.java
package scenarios.cassandracodecs;

import com.datastax.oss.driver.api.core.type.codec.MappingCodec;
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
import com.datastax.oss.driver.api.core.type.reflect.GenericType;

/**
 * Holds a private nested {@code TypeCodec} that cannot be reached from any package.
 * Used to verify the visibility check rejects unconditionally inaccessible classes.
 */
public final class NonAccessibleCassandraCodec {

    private NonAccessibleCassandraCodec() {
    }

    private static final class Inner extends MappingCodec<String, String> {

        public Inner() {
            super(TypeCodecs.TEXT, GenericType.of(String.class));
        }

        @Override
        protected String innerToOuter(String value) {
            return value;
        }

        @Override
        protected String outerToInner(String value) {
            return value;
        }
    }
}

File: ksp/src/test/resources/scenarios/cassandracodecs/NonAccessibleCassandraCodec.kt
package scenarios.cassandracodecs

import com.datastax.oss.driver.api.core.type.codec.MappingCodec
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs
import com.datastax.oss.driver.api.core.type.reflect.GenericType

/**
 * Top-level `private` (file-scoped) class. Cannot be referenced from any other file,
 * regardless of module or package. Used to verify that the visibility check rejects
 * unconditionally inaccessible classes.
 */
private class NonAccessibleCassandraCodec :
    MappingCodec<String, String>(TypeCodecs.TEXT, GenericType.of(String::class.java)) {
    override fun innerToOuter(value: String?): String? = value
    override fun outerToInner(value: String?): String? = value
}
